sensorprocess/sensorprocess.go

```diff
@@ -109,3 +112,96 @@ func tryAddSensorReading(ctx context.Context, reading []byte, readingTime time.T
 	timeElapsedMs := int(time.Since(startTime).Milliseconds())
 	return int(math.Max(0, float64(config.LidarDataRateMsec-timeElapsedMs)))
 }
+
+// StartIMU polls the IMU to get the next sensor reading and adds it to the cartofacade.
+// stops when the context is Done.
+func StartIMU(
+	ctx context.Context,
+	config Config,
+) bool {
+	for {
+		select {
+		case <-ctx.Done():
+			return false
+		default:
+			if jobDone := addIMUReading(ctx, config); jobDone {
+				return true
+			}
+		}
+	}
+}
+
+// addLidarReading adds a lidar reading to the cartofacade.
+func addIMUReading(
+	ctx context.Context,
+	config Config,
+) bool {
+	tsr, err := config.IMU.TimedIMUSensorReading(ctx)
+	if err != nil {
+		config.Logger.Warn(err)
+		// only end the sensor process if we are in offline mode
```

Comment:
[q] @nicksanford @PStrutz how do we want to determine offline mode for IMUs that are optional? Based on the Lidar data rate as seen below?

Reply:
lets talk about this in team-slam-dev

Reply:
Determine based on lidar data rate, if lidar data rate is 0 and imu is not, error